File: docs/demo/V4.x/TESS_JavaAPI_demo/Intelligent_Connected_and_Autonomous_Driving/Connected_and_Automated_Vehicle_Merging_Decision/MyPlugin.java
package TESS_Java_APIDemo.Intelligent_Connected_and_Autonomous_Driving.Connected_and_Automated_Vehicle_Merging_Decision;
import com.jidatraffic.tessng.CustomerNet;
import com.jidatraffic.tessng.CustomerSimulator;
import com.jidatraffic.tessng.TessPlugin;

public class MyPlugin extends TessPlugin
{
    private MyNet mNet;
    private MySimulator mSimu;

    public MyPlugin()
    {
        super();
    }

    @Override
    public void init() {
        mNet = new MyNet();
        mSimu = new MySimulator();
    }

    @Override
    public CustomerSimulator customerSimulator() {
        return mSimu;
    }

    @Override
    public CustomerNet customerNet() {
        return mNet;
    }
}

File: docs/demo/V4.x/TESS_CppAPI_DEMO/5.智能网联与自动驾驶/智能网联车辆汇入决策/MyPlugin.cpp
#include "MyPlugin.h"

#include <QDockWidget>
#include <QMessageBox>

#include "MyNet.h"
#include "MySimulator.h"

MyPlugin::MyPlugin()
	:mpMyNet(nullptr), mpMySimulator(nullptr)
{
}

void MyPlugin::init()
{
	mpMyNet = new MyNet();
	mpMySimulator = new MySimulator();
}

void MyPlugin::unload() {
	delete mpMyNet;
	delete mpMySimulator;
}

CustomerGui *MyPlugin::customerGui()
{
	return nullptr;
}

CustomerNet *MyPlugin::customerNet()
{
	return mpMyNet;
}

CustomerSimulator *MyPlugin::customerSimulator()
{
	return mpMySimulator;
}

MyPlugin::~MyPlugin()
{
}